core/arch/aarch64/emit_utils.c

```diff
@@ -891,14 +891,23 @@ emit_indirect_branch_lookup(dcontext_t *dc, generated_code_t *code, byte *pc,
      * will be restored by the fragment prefix.
      */
 
-    /* Recover app's original x2. */
-    APP(&ilist, instr_create_restore_from_tls(dc, DR_REG_R2, TLS_REG2_SLOT));
-
     /* ldr x0, [x1, #start_pc_fragment_offset] */
     APP(&ilist,
         INSTR_CREATE_ldr(dc, opnd_create_reg(DR_REG_X0),
                          OPND_CREATE_MEMPTR(
                              DR_REG_X1, offsetof(fragment_entry_t, start_pc_fragment))));
+
+    /* Save next_tag value from r2 to r1. We may need to store it into dcontext in the
```

Comment:
This code was modeled on ARM I would guess: did you check whether ARM has the same problem?  At first glance, it looks like it does.

Reply:
It seems so too, but I haven't tested it on an ARM machine. The ibl-stress test isn't ported to ARM yet, and I don't have access to an ARM machine. I added the same fix to arm/emit_utils with an ASSERT_NOT_TESTED. Does that work for now?

Reply:
SGTM